mixer/adapter/rbac/rbac.go

```diff
@@ -15,198 +15,38 @@
-		cacheDuration: b.adapterConfig.CacheDuration,
-		store:         s,
-		closing:       make(chan bool),
-		done:          make(chan bool),
-	}
-	if err = h.startController(s); err != nil {
-		return nil, env.Logger().Errorf("Unable to start controller: %v", err)
-	}
-
-	return h, nil
-}
-
-// maxEvents is the likely maximum number of events
-// we can expect in a second. It is used to avoid slice reallocation.
-const maxEvents = 50
-
-var watchFlushDuration = time.Second
-
-// startController creates a controller from the given params.
-func (h *handler) startController(s store.Store) error {
-	data, watchChan, err := startWatch(s)
-	if err != nil {
-		return h.env.Logger().Errorf("Error while starting watching CRDs: %v", err)
-	}
-
-	c := &controller{
-		configState: data,
-		rbacStore:   h.rbac.(*ConfigStore),
-	}
-
-	c.processRBACRoles(h.env)
-	// Start a goroutine to watch for changes on a channel and
-	// publishes a batch of changes via applyEvents.
-	h.env.ScheduleDaemon(func() {
-		// consume changes and apply them to data.
-		var timeChan <-chan time.Time
-		var timer *time.Timer
-		events := make([]*store.Event, 0, maxEvents)
-
-		for {
-			select {
-			case ev := <-watchChan:
-				if len(events) == 0 {
-					timer = time.NewTimer(watchFlushDuration)
-					timeChan = timer.C
-				}
-				events = append(events, &ev)
-			case <-timeChan:
-				timer.Stop()
-				timeChan = nil
-				h.env.Logger().Infof("Publishing %d events", len(events))
-				c.applyEvents(events, h.env)
-				events = events[:0]
-			case <-h.closing:
-				if timer != nil {
-					timer.Stop()
-					timeChan = nil
-				}
-				h.done <- true
-				return
-			}
-		}
-	})
-	return nil
-}
-
-// startWatch registers with store, initiates a watch, and returns the current config state.
-func startWatch(s store.Store) (map[store.Key]*store.Resource, <-chan store.Event, error) {
-	kindMap := make(map[string]proto.Message)
-	kindMap[serviceRoleKind] = &rbacproto.ServiceRole{}
-	kindMap[serviceRoleBindingKind] = &rbacproto.ServiceRoleBinding{}
-
-	if err := s.Init(kindMap); err != nil {
-		return nil, nil, err
-	}
-	// create channel before listing.
-	watchChan, err := s.Watch()
-	if err != nil {
-		return nil, nil, err
-	}
-	return s.List(), watchChan, nil
-}
-
-////////////////// Request-time Methods //////////////////////////
-// authorization.Handler#HandleAuthorization
-func (h *handler) HandleAuthorization(ctx context.Context, inst *authorization.Instance) (adapter.CheckResult, error) {
-	s := status.OK
-	result, err := h.rbac.CheckPermission(inst, h.env.Logger())
-	if !result || err != nil {
-		s = status.WithPermissionDenied("RBAC: permission denied.")
-	}
-	return adapter.CheckResult{
-		Status:        s,
-		ValidDuration: h.cacheDuration,
-		ValidUseCount: 1000000000,
-	}, nil
-}
-
-// adapter.Handler#Close
-func (h *handler) Close() error {
-	h.closing <- true
-	close(h.closing)
-
-	<-h.done
-	h.store.Stop()
-	return nil
+	return nil, env.Logger().Errorf(prompt)
```

Comment:
I think you need to return success here, otherwise the operator will get errors after upgrade.

Reply:
But I thought that's our intention? To notify the operator to upgrade to proxy based RBAC?
When I tried this locally, it doesn't crash mixs but only output an error message in the log.